sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricQueryResults.java

```diff
@@ -18,24 +18,68 @@
 package org.apache.beam.sdk.metrics;
 
 import com.google.auto.value.AutoValue;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.List;
 import org.apache.beam.sdk.annotations.Experimental;
 import org.apache.beam.sdk.annotations.Experimental.Kind;
+import org.apache.beam.vendor.grpc.v1p13p1.com.google.common.collect.ImmutableList;
 
 /**
  * The results of a query for metrics. Allows accessing all of the metrics that matched the filter.
  */
 @AutoValue
 @Experimental(Kind.METRICS)
 public abstract class MetricQueryResults {
-  /** Return the metric results for the getCounters that matched the filter. */
+  /** Return the metric results for the counters that matched the filter. */
   public abstract Iterable<MetricResult<Long>> getCounters();
 
-  /** Return the metric results for the getDistributions that matched the filter. */
+  /** Return the metric results for the distributions that matched the filter. */
   public abstract Iterable<MetricResult<DistributionResult>> getDistributions();
 
-  /** Return the metric results for the getGauges that matched the filter. */
+  /** Return the metric results for the gauges that matched the filter. */
   public abstract Iterable<MetricResult<GaugeResult>> getGauges();
 
+  static <T> void printMetrics(String type, Iterable<MetricResult<T>> metrics, PrintStream ps) {
+    List<MetricResult<T>> metricsList = ImmutableList.copyOf(metrics);
+    if (!metricsList.isEmpty()) {
+      ps.print(type + "(");
+      // ps.println(metricsList.size() + " " + type + ":");
+      boolean first = true;
+      for (MetricResult<T> metricResult : metricsList) {
+        if (first) {
+          first = false;
+        } else {
+          ps.print(", ");
+        }
+        MetricName name = metricResult.getName();
+        ps.print(metricResult.getStep() + ":" + name.getNamespace() + ":" + name.getName());
+        ps.print(": ");
+        ps.print(metricResult.getAttempted());
+        try {
+          T committed = metricResult.getCommitted();
+          ps.print(", ");
+          ps.print(committed);
+        } catch (UnsupportedOperationException ignored) {
+        }
+      }
+      ps.print(")");
+    }
+  }
+
+  @Override
+  public String toString() {
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    PrintStream ps = new PrintStream(baos);
+    ps.print("MetricQueryResults(");
```

Comment:
Can be replaced by `StringBuilder`. I think this is what you want here.

Reply:
Good point, done.

An earlier implementation assembled a big multi-line string, so `PrintStream.println` was handy, but I decided to do a more idiomatic single-line `toString()` instead.